types/schemas/app/index.ts

```diff
@@ -0,0 +1,63 @@
+import {
+  ApplicationTypeKeys,
+  PrimaryApplicationType,
+} from '../digitalPlanningApplication/enums/ApplicationTypes';
+import {ApplicationData} from './ApplicationData';
+import {User} from './User';
+
+/**
+ * @internal
+ * The generic base type for all applications
+ * Takes a primary and granular application type which allows child properties to differ based on these inputs
+ * Deriving `TPrimary` from `TGranular` is possible in TS, but not in a way which is currently compatible with ts-json-schema-generator
+ */
+interface ApplicationSpecification<
+  TPrimary extends PrimaryApplicationType,
+  TGranular extends ApplicationTypeKeys,
+> {
+  applicationType: TGranular;
+  data: {
+    user: User<TPrimary>;
+    application: ApplicationData<TPrimary>;
+  };
+}
+
+export type PlanningPermissionApplication = ApplicationSpecification<
+  'pp',
+  'pp'
+>;
+// TODO: All granular types
+
+export type PriorApprovalApplication = ApplicationSpecification<'pa', 'pa'>;
+// TODO: All granular types
+
+export type WorksToTreesApplications = ApplicationSpecification<
+  'wtt',
+  Extract<ApplicationTypeKeys, 'wtt' | 'wtt.consent' | 'wtt.notice'>
```

Comment:
This is a really neat solution (thanks again @jessicamcinchak!) and again points towards a simpler way of encoding enums, e.g. 

```ts
type WTTApplicationType =
  /**
   * @description Something
   */
  | "wtt"
  /**
   * @description Something else
   */
  | "wtt.consent"
  /**
   * @description Something else again
   */
  | "wtt.notice";

type ApplicationType = WTTApplicationType | PPApplicationType | PAApplicationType;

export type WorksToTreesApplications = ApplicationSpecification<
  'wtt',
  WTTApplicationType
>
```

This would solve the issue of missing or mismatched types, and having to repeat them here, if they were just defined once. I'll take a closer look at this in another PR.